pkg/k8s/scaledobject.go

```
@@ -52,33 +61,30 @@ func NewScaledObject(
 		var vIface interface{} = v
 		labels[k] = vIface
 	}
+
+	tpl, err := template.ParseFS(scaledObjectTemplateFS, "templates/scaledobject.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var scaledObjectTemplateBuffer bytes.Buffer
+	tpl.Execute(&scaledObjectTemplateBuffer, map[string]interface{}{
+		"Name": name,
+		"Namespace": namespace,
+		"Labels": labels,
+		"MinReplicas": minReplicas,
+		"MaxReplicas": maxReplicas,
+		"DeploymentName": deploymentName,
+		"ScalerAddress": scalerAddress,
+	})
+
+	var decodedYaml map[string]interface{}
+	decodeErr := yaml.Unmarshal(scaledObjectTemplateBuffer.Bytes(), &decodedYaml)
+	if decodeErr != nil {
+		log.Fatal(err)
```

@khaosdoctor instead of `log.Fatal`, can you please return the error?